Source/Scene/Batched3DModel3DTileContent.js

```diff
@@ -199,25 +199,49 @@ define([
         var byteLength = view.getUint32(byteOffset, true);
         byteOffset += sizeOfUint32;
 
+        var batchTableJsonByteLength = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+
+        var batchTableBinaryByteLength = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+
         var batchLength = view.getUint32(byteOffset, true);
-        this._featuresLength = batchLength;
         byteOffset += sizeOfUint32;
 
+        // TODO : remove this legacy check before merging into master
+        // Legacy header:  [batchLength] [batchTableByteLength]
+        // Current header: [batchTableJsonByteLength] [batchTableBinaryByteLength] [batchLength]
+        // If the header is in the legacy format 'batchLength' will be the start of the JSON string (a quotation mark) or the glTF magic.
+        // Accordingly the first byte of uint32 will be either 0x22 or 0x67 and so the uint32 will exceed any reasonable 'batchLength'.
+        if (batchLength > 10000000) {
+            byteOffset -= sizeOfUint32;
+            batchLength = batchTableJsonByteLength;
+            batchTableJsonByteLength = batchTableBinaryByteLength;
+            batchTableBinaryByteLength = 0;
+            console.log('Warning: b3dm header is using the legacy format [batchLength] [batchTableByteLength]. Please update to the latest b3dm format: https://github.com/AnalyticalGraphicsInc/3d-tiles/blob/master/TileFormats/Batched3DModel/README.md.');
```

Comment:
Add the new "[batchTableJsonByteLength] [batchTableBinaryByteLength] [batchLength]" to the string so it is more obvious what changed.